XXX leaving radar coverage without being cleared to XXX

Every departure flight shows "XXX leaving radar coverage without being cleared to XXX", how to solve?

@n8rzz 

Your environment has been set up for using Node.js 7.10.0 (x64) and npm.

C:\Users\12713>npm install
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\12713\package.json'
npm WARN 12713 No description
npm WARN 12713 No repository field.
npm WARN 12713 No README data
npm WARN 12713 No license field.

C:\Users\12713>npm run build
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
npm ERR! node v7.10.0
npm ERR! npm  v4.2.0
npm ERR! path C:\Users\12713\package.json
npm ERR! code ENOENT
npm ERR! errno -4058
npm ERR! syscall open

npm ERR! enoent ENOENT: no such file or directory, open 'C:\Users\12713\package.json'
npm ERR! enoent ENOENT: no such file or directory, open 'C:\Users\12713\package.json'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\12713\AppData\Roaming\npm-cache\_logs\2017-05-21T02_26_47_311Z-debug.log

C:\Users\12713>npm run start
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "start"
npm ERR! node v7.10.0
npm ERR! npm  v4.2.0
npm ERR! path C:\Users\12713\package.json
npm ERR! code ENOENT
npm ERR! errno -4058
npm ERR! syscall open

npm ERR! enoent ENOENT: no such file or directory, open 'C:\Users\12713\package.json'
npm ERR! enoent ENOENT: no such file or directory, open 'C:\Users\12713\package.json'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\12713\AppData\Roaming\npm-cache\_logs\2017-05-21T02_26_57_359Z-debug.log

@n8rzz How to run `git clone http://github.com/openscope/opscope.git` and `cd openscope/`?

C:\Users\12713\Desktop\openscope-develop>npm install
npm WARN deprecated gulp-minify-css@1.2.4: Please use gulp-clean-css
npm ERR! git clone --template=C:\Users\12713\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:jonschlinkert/micromatch.git C:\Users\12713\AppData\Roaming\npm-cache\_git-remotes\git-github-com-jonschlinkert-micromatch-git-2-2-0-c0994e2c: undefined
npm ERR! git clone --template=C:\Users\12713\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:jonschlinkert/micromatch.git C:\Users\12713\AppData\Roaming\npm-cache\_git-remotes\git-github-com-jonschlinkert-micromatch-git-2-2-0-c0994e2c: undefined
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v7.10.0
npm ERR! npm  v4.2.0
npm ERR! code ENOGIT

npm ERR! not found: git
npm ERR!
npm ERR! Failed using git.
npm ERR! This is most likely not a problem with npm itself.
npm ERR! Please check if you have git installed and in your PATH.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\12713\AppData\Roaming\npm-cache\_logs\2017-05-21T02_46_47_310Z-debug.log

@n8rzz 

```
$ git clone http://github.com/openscope/openscope.git
Cloning into 'openscope'...
warning: redirecting to https://github.com/openscope/openscope.git/
remote: Counting objects: 26698, done.
remote: Compressing objects: 100% (90/90), done.
remote: Total 26698 (delta 46), reused 75 (delta 25), pack-reused 26582
Receiving objects: 100% (26698/26698), 13.63 MiB | 872.00 KiB/s, done.
Resolving deltas: 100% (17540/17540), done.
```
Then run `npm install` in Node.js but failed.
```
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v7.10.0
npm ERR! npm  v4.2.0
npm ERR! code ENOGIT

npm ERR! not found: git
npm ERR!
npm ERR! Failed using git.
npm ERR! This is most likely not a problem with npm itself.
npm ERR! Please check if you have git installed and in your PATH.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\12713\AppData\Roaming\npm-cache\_logs\2017-05-21T03_33_35_404Z-debug.log
```

It is too difficult for non-professionals like me to install, (⊙﹏⊙)b

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,7 +16,7 @@
 - Adds `StripViewController`, `StripViewCollection`, and `StripViewModel` classes [#285](https://github.com/openscope/openscope/issues/285)
     - Removes progress strip logic from the `AircraftModel`
     - Completely reworks CSS for `StripViewList`
-
+- Updates `AircraftModel.onAirspaceExit()` to look only at the `mcp.headingMode` value [#477](https://github.com/openscope/openscope/issues/477)
 
 
 

--- a/src/assets/scripts/client/aircraft/AircraftInstanceModel.js
+++ b/src/assets/scripts/client/aircraft/AircraftInstanceModel.js
@@ -405,10 +405,9 @@ export default class AircraftInstanceModel {
 
         this.setIsRemovable();
 
-        // TODO: this seems redundant. if its already in the leg its in the fms.
-        if (this.mcp.headingMode !== MCP_MODE.HEADING.LNAV || !this.fms.hasWaypoint(this.fms.currentLeg.exitName)) {
+        if (this.mcp.headingMode !== MCP_MODE.HEADING.LNAV) {
             this.radioCall(
-                `leaving radar coverage without being cleared to ${this.fms.currentLeg.exitName}`,
+                'leaving radar coverage without proper clearance',
                 AIRPORT_CONTROL_POSITION_NAME.DEPARTURE,
                 true
             );

